### PR TITLE
Search nested select

### PR DESCRIPTION
**Pull request by Wilker0725**
**Changes**
Description of changes...
* Implement search function

- [delete] branch

- [WARNING] Description of change in `NestedSelect`

- [Figma](https://www.figma.com/file/us90jOpWBahsqK2VAluPvD/Flagship-Design-System?node-id=11728%3A111640&mode=dev)

### CODE REVIEW

**Review by qroll — CHANGES_REQUESTED** [review]

File: src/shared/nested-dropdown-list/nested-dropdown-list-helper.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                });
+            }
+        );
+
+        return list;
+    };
+
+    export const getVisibleKeyPaths = <V1, V2, V3>(
+        list: Map<string, CombinedFormattedOptionProps<V1, V2, V3>>,
+        isSearch?: boolean
```
Comment (qroll) [inline src/shared/nested-dropdown-list/nested-dropdown-list-helper.ts:37]: is it intended for the keyPaths to be out of order in search? I'm wondering if you can remove this isSearch flag entirely

File: src/shared/nested-dropdown-list/types.ts [hunk 2]
```diff
@@ -12,9 +12,19 @@ export interface DropdownEventHandlerProps<V1, V2, V3> {
...
 }
 
+export interface DropdownSearchProps {
+    /** Specifying will render a search bar in the dropdown */
+    enableSearch?: boolean | undefined;
+    /** If specified, the default no results display will not be rendered */
+    hideNoResultsDisplay?: boolean | undefined;
+    searchPlaceholder?: string | undefined;
+    onSearch?: (() => void) | undefined;
+}
```
Comment (qroll) [inline src/shared/nested-dropdown-list/types.ts:22]: to pass `onSearch` and `hideNoResultsDisplay` to the nested dropdown list

File: src/shared/nested-dropdown-list/list-item.styles.tsx
Comment (qroll) [inline src/shared/nested-dropdown-list/list-item.styles.tsx:1]: Add a `overflow-wrap: anywhere;` to `Title` so that the text can break

**Review by qroll — CHANGES_REQUESTED** [review]

File: src/shared/nested-dropdown-list/nested-dropdown-list-helper.ts (hunk 1 above)
Comment (Wilker0725) [reply src/shared/nested-dropdown-list/nested-dropdown-list-helper.ts:37]: Thanks spotted this one!!!

**Review by qroll — APPROVED** [review]

**Pull request merged**